hnn_core/gui.py

```diff
@@ -0,0 +1,474 @@
+    rate_constant = dict()
+    for cell_type in cell_types:
+        rate_constant[f'{cell_type}'] = FloatText(
+            value=8.5, description=f'{cell_type}:',
+            layout=layout, style=style)
+
+    widgets_list, widgets_dict = _get_cell_specific_widgets(layout, style, location)
+    widgets_dict.update({'rate_constant': rate_constant})
+    widgets_list.extend([HTML(value="<b>Rate constants</b>")] +
+                        list(widgets_dict['rate_constant'].values()))
+
+    drive_box = VBox([tstart, tstop, seedcore] + widgets_list)
+    drive = dict(type='Poisson', name=name, tstart=tstart,
+                 tstop=tstop, rate_constant=rate_constant,
+                 seedcore=seedcore,
+                  )
+    drive.update(widgets_dict)
+    return drive, drive_box
+
+
+def _get_evoked_widget(name, layout, style, location):
+    kwargs = dict(layout=layout, style=style)
+    mu = FloatText(value=0, description='Mean time:', **kwargs)
+    sigma = FloatText(value=1, description='Std dev time:',
+                      **kwargs)
+    numspikes = IntText(value=1, description='No. Spikes:',
+                        **kwargs)
+    seedcore = IntText(value=14, description='Seed: ', **kwargs)
+
+    widgets_list, widgets_dict = _get_cell_specific_widgets(layout, style, location)
+
+    drive_box = VBox([mu, sigma, numspikes, seedcore] +
+                     widgets_list)
+    drive = dict(type='Evoked', name=name,
+                 mu=mu, sigma=sigma, numspikes=numspikes,
+                 seedcore=seedcore,
+                 sync_within_trial=False)
+    drive.update(widgets_dict)
+    return drive, drive_box
+
+
+def add_drive_widget(drive_type, drive_boxes, drive_widgets,
+                     drives_out, tstop_widget, location):
+    """Add a widget for a new drive."""
+    layout = Layout(width='270px', height='auto')
+    style = {'description_width': '150px'}
+    drives_out.clear_output()
+    with drives_out:
+        name = drive_type + str(len(drive_boxes))
+
+        if drive_type == 'Rhythmic':
+            drive, drive_box = _get_rhythmic_widget(name, tstop_widget,
+                                                    layout, style, location)
+        elif drive_type == 'Poisson':
+            drive, drive_box = _get_poisson_widget(name, tstop_widget,
+                                                   layout, style, location)
+        elif drive_type == 'Evoked':
+            drive, drive_box = _get_evoked_widget(name, layout, style, location)
+
+        if drive_type in ['Evoked', 'Poisson', 'Rhythmic']:
+            drive_boxes.append(drive_box)
+            drive_widgets.append(drive)
+
+        accordion = Accordion(children=drive_boxes,
+                              selected_index=len(drive_boxes) - 1)
+        for idx, drive in enumerate(drive_widgets):
+            accordion.set_title(idx, drive['name'])
+        display(accordion)
+
+
+def update_plot_window(variables, plot_out, plot_type):
```

Comment:
This might be a good place to package the plot callables and arguments into dictionaries indexed by the values under `plot_type['new']`?

I think `elif` lines are fine for 2-3 items, but perhaps it will read better with something like
```py
key = plot_type['new']
plot_dict[key]['callable'](**args_dict[key])
```